Sources/UnidirectionalBinding.swift

```diff
@@ -73,29 +73,39 @@ public protocol BindingTargetProtocol: class {
 ///
 /// - parameters:
 ///   - target: A target to be bond to.
-///   - signal: A signal to bind.
+///   - source: A source to bind.
 ///
 /// - returns: A disposable that can be used to terminate binding before the
-///            deinitialization of the target or the signal's `completed`
+///            deinitialization of the target or the source's `completed`
 ///            event.
 @discardableResult
 public func <~
 	<Target: BindingTargetProtocol, Source: BindingSourceProtocol>
 	(target: Target, source: Source) -> Disposable?
 	where Source.Value == Target.Value, Source.Error == NoError
 {
-	let disposable = source.observe(Observer(value: { [weak target] in target?.consume($0) }))
+	// Alter the semantics of `BindingTarget` to not require it to be retained.
+	// This is done here--and not in a separate function--so that all variants
+	// of `<~` can get this behavior.
+	let observer: Observer<Source.Value, NoError>
+	if let target = target as? BindingTarget<Source.Value> {
```

Comment:
IIRC Swift will pick the most specific overload. So instead of a dynamic check, you might as well define an overload: 
```
@discardableResult
  public func <~
  	<Value, Source: BindingSourceProtocol>
  	(target: BindingTarget<Value>, source: Source) -> Disposable?
  	where Source.Value == Target.Value, Source.Error == NoError
```
Though the alternative is still having it defined in the protocol, and overriding it in `BindingTarget`.